text/0018-bee-bundle/0018-bee-bundle.md

```diff
@@ -0,0 +1,171 @@
++ Feature name: `bee-bundle`
++ Start date: 2019-10-02
++ RFC PR: [iotaledger/bee-rfcs#18](https://github.com/iotaledger/bee-rfcs/pull/18)
++ Bee issue: [iotaledger/bee#63](https://github.com/iotaledger/bee/issues/63)
+
+# Summary
+
+The smallest communication unit in the IOTA protocol is the transaction. Everything, including payment settlements and/or plain data, is propagated through the IOTA network in transactions.
+
+A transaction is `2673` trytes and the part available to the user is `2187` trytes. This part holds a signature in case of a payment settlement and plain data otherwise. Since it has a limited size, a user often needs more than one transaction to fulfil his operation, for example signatures with security level `2` or `3` don't fit in a single transaction and user-provided data may exceed the allowance so they need to be fragmented across multiple transactions. Moreover, a transaction may not make sense on its own, for example an input/output transaction alone would change the total amount of the ledger so it has to be paired with another complementary input/output.
+
+For these reasons, transactions have to be processed as a whole, in groups called bundles. A bundle is an atomic operation in the sense that either all or none of its transactions are accepted by the network. Even single transactions are propagated through the network within a bundle making it the only confirmable communication unit of the IOTA protocol.
+
+This RFC proposes ways to create and manipulate a bundle and describe the associated algorithms.
+
+Useful links:
+- [What is a bundle?
+](https://docs.iota.org/docs/getting-started/0.1/introduction/what-is-a-bundle)
+- [Bundles and transactions
+](https://docs.iota.org/docs/dev-essentials/0.1/concepts/bundles-and-transactions)
+
+# Motivation
+
+<!-- TODO -->
+
+# Detailed design
+
+<!-- TODO -->
+
+## Algorithms
+
+<!-- TODO -->
+
+### Bundle hash generation
+
+*Client side and server side operation.*
+
+A bundle hash ties different transactions together. By having this common hash in their `bundle` field, it makes it clear that these transactions should be processed as a whole.
+
+The hash of a bundle is derived from the bundle essence of each of its transactions. The bundle essence of a transaction is composed of the following fields.
+
+| Name          | Size      |
+| ------------- | --------- |
+| address       | 243 trits |
+| value         | 81 trits  |
+| obsolete_tag  | 81 trits  |
+| timestamp     | 27 trits  |
+| current_index | 27 trits  |
+| last_index    | 27 trits  |
+
+The bundle hash is generated with a sponge by iterating through the bundle, from `0` to `last_index`, absorbing the bundle essence of each transaction and eventually squeezing the bundle hash from the sponge.
```

Comment:
sponge function

Reply:
Technically, I'm speaking of a sponge object as seen in the pseudocode, so I think it's fine.